lasagne/layers/shape.py

```diff
@@ -300,3 +302,64 @@ def get_output_for(self, input, **kwargs):
         return padding.pad(input, self.width, self.val, self.batch_ndim)
 
 pad = PadLayer  # shortcut
+
+
+class SliceLayer(Layer):
+    """
+    Slices the input at a specific axis and at specific indices.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    indices : int, list, tuple or slice instance
+        When indices is an int the corresponding index is selected from the
+        axis dimension and the axis dimension is dropped. When indices is list,
+        tuple or slice instance the axis dimension is kept even if it is
+        singleton dimension after slicing. Negative values behaves as in numpy
+        indexing.
+
+    axis : int
+        Specifies the axis from which the indices are selected. Negative values
+        behaves as in numpy indexing.
+
+    Examples
+    --------
+    >>> from lasagne.layers import SliceLayer, InputLayer, get_output
+    >>> in_shp = (2, 3, 2)
+    >>> l_inp = InputLayer(in_shp)
+    >>> l_slice_ax0 = SliceLayer(l_inp, axis=0, indices=0)
+    >>> l_slice_ax1 = SliceLayer(l_inp, axis=1, indices=(1, 3))
+    >>> l_slice_ax2 = SliceLayer(l_inp, axis=-1, indices=-1)
+    >>> x = np.arange(np.prod(in_shp)).reshape(in_shp).astype('float32')
+    >>> get_output(l_slice_ax0, x).eval().shape
+    (3, 2)
+    >>> get_output(l_slice_ax1, x).eval().shape
+    (2, 2, 2)
+    >>> get_output(l_slice_ax2, x).eval().shape
+    (2, 3)
```

Comment:
Note that I edited my comment for the usage example, I think the second version is a bit clearer.